components/busola-migrator/internal/config/config.go

```diff
@@ -0,0 +1,91 @@
+package config
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/vrischmann/envconfig"
+
+	"github.com/kyma-project/kyma/components/busola-migrator/pkg/ptr"
+)
+
+type Config struct {
+	Port           int           `envconfig:"default=80"`
+	TimeoutRead    time.Duration `envconfig:"default=30s"`
+	TimeoutWrite   time.Duration `envconfig:"default=30s"`
+	TimeoutIdle    time.Duration `envconfig:"default=120s"`
+	BusolaURL      string        `envconfig:"default=https://busola.main.hasselhoff.shoot.canary.k8s-hana.ondemand.com"`
+	OIDCIssuerURL  string        `envconfig:"default=https://kyma.accounts.ondemand.com"`
+	OIDCClientID   string        `envconfig:"default=6667a34d-2ea0-43fa-9b13-5ada316e5393"`
+	OIDCScope      string        `envconfig:"default=openid"`
+	OIDCUsePKCE    bool          `envconfig:"default=false"`
+	StaticFilesDIR string        `envconfig:"optional"`
+}
+
+type configOverrides struct {
+	BusolaURL     *string
+	OIDCIssuerURL *string
+	OIDCClientID  *string
+	OIDCScope     *string
+	OIDCUsePKCE   *bool
+}
+
+func LoadConfig() Config {
+	var cfg Config
+	err := envconfig.InitWithPrefix(&cfg, "APP")
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "while processing environment variables"))
+	}
+
+	overrides := getOverrides()
+
+	cfg = applyOverrides(cfg, overrides)
+
+	return cfg
+}
+
+func getOverrides() configOverrides {
+	var overrides configOverrides
+
+	if val, ok := os.LookupEnv("OVERRIDE_BUSOLA_URL"); ok {
+		overrides.BusolaURL = ptr.String(val)
```

Comment:
Why you set this temp struct `config Overrides`? Can we set struct `Config` here?

Reply:
I didn't use Config struct and `envconfig` library because it doesn't allow nil values in case of unset variables, I needed a way to distinguish when the environment variable was set to empty string from when it wasn't set at all.